http/logical.go

```diff
@@ -170,7 +170,9 @@ func respondLogical(w http.ResponseWriter, r *http.Request, path string, dataOnl
 				},
 			}
 		} else {
-			httpResp = logical.SanitizeResponse(resp)
+			sanitizedHttp := logical.SanitizeResponse(resp)
```

Comment:
I received an error in my attempt to set requestID directly because httpResp is an interface with no methods. If this isn't the way to handle it, I can change it.